frontend_tests/node_tests/muting.js

```diff
@@ -55,6 +55,36 @@ var muting = require('js/muting.js');
     ]);
 }());
 
+(function test_muting_performance() {
+    // This test ensures that each call to mute_topic and set_muted_topics only
+    // results in one call to unread.update_unread_counts.
+
+    // We replace (for the duration of this test) the real update_unread_counts
+    // with a test version that just counts the number of calls.
+    var saved = unread.update_unread_counts;
+    var num_calls = 0;
+    unread.update_unread_counts = function () {
+        num_calls += 1;
+    };
+
+    muting.mute_topic('office', 'gossip');
+    assert.equal(num_calls, 1)
+
+    muting.set_muted_topics([
+        ['social', 'breakfast'],
+    ]);
+    assert.equal(num_calls, 2)
+
+    muting.set_muted_topics([
+        ['social', 'breakfast'],
+        ['design', 'typography'],
+        ['devel', 'java'],
+    ]);
+    assert.equal(num_calls, 3)
+
+    unread.update_unread_counts = saved;
```

Comment:
Very nice tests!

FYI you are technically able to just let the test-engine code deal with the unread monkeypatches after this module gets run, but I'd leave the line in here to show your intention that's it's just a temporary patch.